### ISSUE TITLE
[feature request] dropout1d

### ISSUE THREAD

**Issue by ruotianluo** [pytorch/pytorch#6442]
While there are dropout2d and dropout3d, there is no dropout1d. And it seems like FeatureDropout naturally supports NxCxL input, should be no effort?

**Comment by Lyken17** [comment]
I think `nn.Dropout` already supports that

```python
In [5]: model = nn.Sequential(
   ...:     nn.Linear(10, 5),
   ...:     nn.Dropout(0.5),
   ...:     nn.Linear(5, 1)
   ...: )

In [10]: data = Variable(torch.ones(5, 3, 10))
In [11]: model(data)
Out[11]:
Variable containing:
(0 ,.,.) =
 -0.1277
 -0.5975
 -1.0493

(1 ,.,.) =
 -0.5795
 -0.1277
 -0.8607

(2 ,.,.) =
 -0.8497
 -0.5865
 -0.8607

(3 ,.,.) =
 -0.8623
 -0.5990
 -0.8607

(4 ,.,.) =
 -0.1403
 -0.5780
 -0.3994
[torch.FloatTensor of size 5x3x1]
```

**Comment by ruotianluo** [comment]
@Lyken17 Dropoutxd dropout the along channels while dropout treat each point individually. @zou3519 I can make a PR if you want.

**Comment by zou3519** [comment]
@ruotianluo I think it's better if we simplify the API. As you mentioned, `FeatureDropout` already supports `(N, C, L)` inputs (and is used to implement `Dropout{2, 3}d`), so it might be better to deprecate the `Dropoutxd` functions and expose `FeatureDropout`. What do you think?

**Comment by ruotianluo** [comment]
I feel like dropoutxd is more straightforward to me. And all the `BatchNormxd`s are also the same at the backend.

**Comment by davidaknowles** [comment]
Did either of these options get implemented?

**Comment by nexplor** [comment]
For the benefit of anyone else who wants to use `Dropout1d` while this is not resolved:
You can just use `Dropout2d` instead, and it will do the exact same thing, because all `dropoutNd` should simply call feature_dropout under the hood anyway

(You can check that the code for `nn.functional.dropout2d` and `nn.functional.dropout3d` is exactly the same)
